### PR TITLE
Zero the FFT output when the input is empty but fft_length is not

### LINKED ISSUE #126898: tf.signal.rfft returns non-zero / non-finite values for zero-length input axis with positive explicit fft_length
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

 Windows

### Mobile device

_No response_

### Python version

3.11

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

tf.signal.rfft(tf.zeros((2, 2, 0), dtype=tf.float32), fft_length=[16])`
should return a `complex64` tensor with shape `[2, 2, 9]`. Since the input is
empty along the transform axis and the positive explicit FFT length pads with
zeros, all output values should be finite zeros.

An alternative acceptable behavior would be a clear validation error if
TensorFlow intentionally does not support this boundary case. Returning
non-zero or non-finite values appears incorrect.

The result shape is `[2, 2, 9]`, but some runs contain non-zero and/or
non-finite values, including `nan+nanj`. The failure is intermittent within a
single process, which looks like uninitialized output memory.

### Standalone code to reproduce the issue

```shell
import os
import platform
import tensorflow as tf

TRIALS = 100
EXPECTED_SHAPE = [2, 2, 9]

print("python:", platform.python_version())
print("platform:", platform.platform())
print("tensorflow:", tf.__version__)
print("TF_ENABLE_ONEDNN_OPTS:", os.environ.get("TF_ENABLE_ONEDNN_OPTS"))

bad_shape = 0
nonfinite = 0
nonzero = 0
first_bad = None

for trial in range(TRIALS):
    y = tf.signal.rfft(
        tf.zeros((2, 2, 0), dtype=tf.float32),
        fft_length=[16],
    )

    real = tf.math.real(y)
    imag = tf.math.imag(y)
    is_finite = bool(
        tf.reduce_all(
            tf.logical_and(tf.math.is_finite(real), tf.math.is_finite(imag))
        ).numpy()
    )
    nonzero_count = int(
        (tf.math.count_nonzero(real) + tf.math.count_nonzero(imag)).numpy()
    )

    if y.shape.as_list() != EXPECTED_SHAPE:
        bad_shape += 1
    if not is_finite:
        nonfinite += 1
    if nonzero_count:
        nonzero += 1
    if first_bad is None and (not is_finite or nonzero_count):
        first_bad = {
            "trial": trial,
            "shape": y.shape.as_list(),
            "first_values": y.numpy().reshape(-1)[:10].tolist(),
        }

print(
    {
        "trials": TRIALS,
        "expected_shape": EXPECTED_SHAPE,
        "bad_shape": bad_shape,
        "nonfinite": nonfinite,
        "nonzero": nonzero,
        "first_bad": first_bad,
    }
)
```

### Relevant log output

```shell
{'trials': 100, 'expected_shape': [2, 2, 9], 'bad_shape': 0, 'nonfinite': 7, 'nonzero': 7, 'first_bad': {'trial': 1, 'shape': [2, 2, 9], 'first_values': [0j, 0j, 0j, 0j, (-2.181645667963854e+30+4.5914945482066956e-41j), 0j, (1.7796490496925177e-43+0j), (2.677369972350537e+36+8.169570047013684e-43j), (2.677537302229767e+36+8.169570047013684e-43j), (2.6777198439162e+36+8.169570047013684e-43j)]}}
```

### PR DESCRIPTION
**Pull request by vishwakt**
Fixes #126898.

## Problem

`tf.signal.rfft(tf.zeros((2, 2, 0)), fft_length=[16])` returns a `[2, 2, 9]` tensor whose contents are intermittently garbage (denormals, huge values, `nan`). The output is never written.

The empty-input early return shared by `FFTBase::Compute` and `FFTNBase::Compute` assumes an empty input implies an empty output, and says so:

```cpp
if (input_shape.num_elements() == 0) {
  DCHECK_EQ(0, output_shape.num_elements());
  return;
}
```

With an explicit `fft_length` the output shape comes from `fft_length`, so the output can be non-empty while the input is empty. The `DCHECK` is compiled out in release builds, so the kernel returns the freshly allocated output without ever writing it. The Python wrapper deliberately skips padding empty tensors (`_maybe_pad_for_rfft`, "Edge case: skip padding empty tensors"), which is how the empty input reaches the kernel through the public API; `tf.raw_ops` callers reach it directly, so the kernel is the right place to fix it.

Reproduced on TF 2.21.0 (CPU): for `rfft2d` on a `[2, 0, 0]` input with `fft_length=[8, 16]`, 142 of the 144 output values are stale memory such as `2.76e+20+1.79e+31j`; the 1-D case in the report reproduces intermittently depending on what the allocator hands back, matching the reporter's 7 of 100 trials.

## Fix

Such an output is the transform of pure zero padding, so it is all zeros. When the input is empty and the output is not, fill the output with zeros instead of returning it uninitialized. The zeroing goes through a new `ZeroOutput` virtual implemented per device, since the base classes serve both the DUCC CPU kernel (`memset`) and the cuFFT kernels (`Stream::MemZero`); cuFFT could not have been handed the empty input directly, as its plan would receive a zero `inembed`, so removing the early return was not an option. The truly empty-output case keeps its early return.

## Test

`testEmptyInputWithFftLengthIsZero` in `fft_ops_test.py` covers `rfft`, `rfft2d`, and `irfft` on inputs that are empty along the transform axes with a positive `fft_length`, asserting all-zero output of the expected shape and dtype. Against the current wheel it fails in both graph and eager mode on the `rfft2d` case with the stale buffer contents above, and the test target is a CUDA test, so CI also exercises the `MemZero` path.

### CODE REVIEW

**Review by dmiltr3 — APPROVED** [review]
I have verified that your changes cleanly apply and pass all tests locally. The failing `build-windows-x86` check on GitHub Actions is a known infrastructure flake and is not related to your code. Your PR is technically sound and is now Ready for Merge.
